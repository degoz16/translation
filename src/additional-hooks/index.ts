import {
    CustomRendererHook,
    CustomRendererHooks,
    CustomRendererLifeCycle,
} from '@diplodoc/markdown-it-custom-renderer';

export type AdditionalHooks = Partial<
    Record<keyof typeof CustomRendererLifeCycle, CustomRendererHook | CustomRendererHook[]>
>;
const lifeCycleStringToEnums = (
    lifeCycle: keyof typeof CustomRendererLifeCycle,
): CustomRendererLifeCycle => {
    return CustomRendererLifeCycle[lifeCycle];
};

export const mergeAdditionalHooks = (
    defaultHooks: CustomRendererHooks,
    additionalHooks?: CustomRendererHooks,
) => {
    const result: CustomRendererHooks = {};
    if (!additionalHooks) {
        return defaultHooks;
    }

    const enumAddtionalHooks = Object.keys(additionalHooks).reduce(
        (acc: CustomRendererHooks, key: string) => {
            acc[lifeCycleStringToEnums(key as keyof typeof CustomRendererLifeCycle)] =
                additionalHooks[key];
            return acc;
        },
        {},
    );

    const keys = new Set([...Object.keys(defaultHooks), ...Object.keys(enumAddtionalHooks)]);
    for (const key of keys) {
        result[key] = [defaultHooks[key], enumAddtionalHooks[key]].flat().filter((hook) => hook);
    }
    return result;
};
