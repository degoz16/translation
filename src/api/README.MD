# api

api that allows xlf and skeleton generation from markdown, and composition of the markdown from xlf and skeleton.

## Overview

- [extract](#extract)
- [compose](#compose)

## extract

generate xlf and skeleton from markdown

## compose

compose markdown from xlf and skeleton
