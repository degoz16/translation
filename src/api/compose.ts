function compose() {}

export {compose};
export default {compose};
