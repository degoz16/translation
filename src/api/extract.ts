import skeleton from 'src/skeleton';

export type ExtractParameters = skeleton.renderer.RenderParameters;

export type ExtractOutput = {
    skeleton: string;
    xlf: string;
};

function extract(parameters: ExtractParameters): ExtractOutput {
    // upon recieving empty markdown give back empty xlf and skeleton
    if (!parameters.markdown) {
        return {xlf: '', skeleton: ''};
    }

    // proper defaults for markdown and skeleton ids
    if (!parameters.markdownPath) {
        // eslint-disable-next-line no-param-reassign
        parameters.markdownPath = 'markdown.md';
    }

    if (!parameters.skeletonPath) {
        // eslint-disable-next-line no-param-reassign
        parameters.skeletonPath = 'markdown.skl.md';
    }

    if (!validParameters(parameters)) {
        throw new Error('invalid parameters');
    }

    return skeleton.renderer.render(parameters);
}

function validParameters(parameters: ExtractParameters) {
    return skeleton.renderer.validParameters(parameters);
}

export {extract};
export default {extract};
