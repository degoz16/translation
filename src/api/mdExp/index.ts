export type {
    ExtractOptions,
    ExtractOutput,
    ComposeOptions,
    ComposeOutput,
} from 'src/experiment/adapter/types';

export {extract, compose} from 'src/experiment/adapter/adapter';
