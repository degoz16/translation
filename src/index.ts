export {extract, ExtractParams} from './api/extract';
export {compose, ComposeParams} from './api/compose';

export * as default from './api';
