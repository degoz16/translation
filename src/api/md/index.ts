import type {TemplateOptions} from 'src/xliff'
import {hash} from 'src/hash';
import {render} from 'src/skeleton';
import {fromXLIFF, parse, template} from 'src/xliff';
import {replace} from 'src/utils';

export type ExtractOptions = TemplateOptions;

export type ExtractOutput = {
    skeleton: string;
    xliff: string;
    units: string[];
};

export type ComposeOptions = {
    useSource?: boolean;
};

export function extract(content: string, options: ExtractOptions): ExtractOutput {
    if (!content) {
        return {xliff: '', units: [], skeleton: ''};
    }

    const hashed = hash();

    return {
        skeleton: render(content, hashed),
        xliff: template(hashed.segments, options),
        units: hashed.segments,
    };
}

export function compose(skeleton: string, xliff: string | string[], {useSource = false}) {
    const units = parse(xliff, {useSource}).map(fromXLIFF);

    return replace(skeleton, units)[0];
}

