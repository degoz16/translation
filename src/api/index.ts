import type {ComposeOptions as MdComposeOptions, ExtractOptions as MdExtractOptions, ExtractOutput as MdExtractOutput } from './md';
import type {ComposeOptions as JsonComposeOptions, ExtractOptions as JsonExtractOptions, ExtractOutput as JsonExtractOutput } from './json';
import {compose as composeMd, extract as extraactMd } from './md';
import {compose as composeJson, extract as extraactJson} from './json';
import {validate} from './validate';
import {JSONValue} from 'src/json';

export type ExtractOptions = Parameters<typeof extract>[1];

export type ComposeOptions = Parameters<typeof compose>[2];

export function extract(content: JSONValue, params: JsonComposeOptions): JsonExtractOutput
export function extract(content: string, params: MdExtractOptions): MdExtractOutput
export function extract(content: any, params: any): any {
    validate('ExtractOptions', params);

    const type = typeof content === 'string' ? 'md' : 'json';

    return extract[type](content, params);
}

extract.md = extraactMd;
extract.json = extraactJson;

export function compose(skeleton: JSONValue, xliff: string | string[], options: JsonExtractOptions): JSONValue
export function compose(skeleton: string, xliff: string | string[], options: MdComposeOptions): string
export function compose(skeleton: any, xliff: string | string[], options: any): any {
    validate('ComposeOptions', options);

    const type = typeof skeleton === 'string' ? 'md' : 'json';

    return compose[type](skeleton, xliff, options);
}

compose.md = composeMd;
compose.json = composeJson;