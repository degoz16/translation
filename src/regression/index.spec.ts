/* eslint-disable @typescript-eslint/no-unused-expressions */

import {compose, extract} from 'src/api';

const getPadX = (string: string) => {
    const match = /^(\s+)/.exec(string);
    const pad = (match && match[1]) || '';

    // eslint-disable-next-line security/detect-non-literal-regexp
    return new RegExp('^[\\s]{0,' + pad.length + '}');
};

export function trim(string: string | TemplateStringsArray): string {
    let lines = Array.isArray(string) ? (string as string[]) : (string as string).split('\n');

    let pad: RegExp | null = null;

    while (pad === null) {
        const line = lines[0] || '';
        if (!line.trim()) {
            lines.shift();
            continue;
        }

        pad = getPadX(line);
    }

    if (pad) {
        lines = lines.map((line) => line.replace(pad as RegExp, ''));
    }

    return lines.join('\n').trim();
}

const test = (() => {
    function test(name: string, call?: 'skip' | 'only') {
        return function (parts: TemplateStringsArray) {
            const markdown = trim(parts.join(''));

            describe('integration', () => {
                const caller = call ? it[call] : it;
                caller(name, () => {
                    const {xliff, units, skeleton} = extract({
                        markdown,
                        source: {
                            language: 'ru',
                            locale: 'RU',
                        },
                        target: {
                            language: 'en',
                            locale: 'US',
                        },
                    });
                    const result = compose({xliff, units, skeleton, useSource: true});

                    // console.log(xliff);

                    expect(xliff).toMatchSnapshot();
                    expect(result).toEqual(markdown);
                    expect(result).toMatchSnapshot();
                });
            });
        };
    }

    test.skip = (name: string) => test(name, 'skip');
    test.only = (name: string) => test(name, 'only');

    return test;
})();

test('link variable leak')`
    ## Request
    
    #||| **method** | **url** ||
    
    ||
    \`\`\`
    POST
    \`\`\`
    | 
    \`\`\`
    https://api.example.ru/businesses/{businessId}/offer-mappings/update
    \`\`\`
    |||#
    
    ### Path parameters
    
    #||| **Name** | **Type** | **Description** ||
    
    || businessId | integer&lt;int64&gt; | Description |||#
`;

test('recursive merge tokens')`
    Sentense [link](file.md "title").
`;

test('handles inline code 1')`
    Параметр | Описание
    ----- | -----
    \`InternalServerError\` | Произошла
`;

// test.skip('handles inline code 2')`
//     The \`Success! The configuration is valid\`
// `;

test('handles inline code 3')`
    Описание значений полей:
    | **Название поля** | **Тип** | **Описание** |
    | --- | --- | --- |
    |column_name|\`string\`<br>\`required\`|Имя колонки для TTL|
`;

test('computes valid sentenses')`
    Записывает или удаляет элементы из таблиц.
    Один вызов может записать до 16 Мб данных, что может включать до 25 запросов на размещение или удаление. Каждый элемент может иметь размер до 400 Кб.
`;

test('handles titleless notes')`
    {% note info %}

    Для работы в режиме совместимости с Amazon DynamoDB используйте бессерверную (Serverless) конфигурацию БД.
    
    {% endnote %}
`;

test('handles blockquotes')`
    > 1
    > 2
    >
    > > 3
    > > 4
    >
    > 5
`;

test('handles multiline inline code')`
    \`ALL_CONCAT( expression [ , separator ]
        [ FIXED ... | INCLUDE ... | EXCLUDE ... ]
        [ BEFORE FILTER BY ... ]
    )\`
`;

test('handles code regexps with escape chars')`
    Поле | Описание
    --- | ---
    zone |  \`\`[.]\\|[a-z0-9][-a-z0-9.]*\\.\`\`
`;

test('handles links in brackets')`
    (see [below for nested schema](#nestedblock--timeouts))
`;

test('handles links with brackets')`
    [^\\[Упр\\]^](#example)
`;

test.skip('handles multiline tabs title')`
    {% list tabs %}

    * Tab1
      Tab1-Line2
    
      Tab content
    
    * Tab2
    
      Tab2-Line2
    
      Tab content
    
    {% endlist %}
`;

test('handles autolinks')`
     * A <http://gitgrimbo.github.io/harviewer/master/>
     * B <https://toolbox.googleapps.com/apps/har_analyzer/?lang=ru>
`;

test('## Start {#your-ideas}{% else %} end')`
    ## Start {#your-ideas}{% else %} end
`;

test('`{{ var }}` — one. Two')`
    \`{{ var }}\` — one. Two
`;

test('(a **18. B**).')`
    (a **18. B**).
`;

test('A ![](link) b. C.')`
    A ![](link) b. C.
`;

test('A (B) {#C}\n:   A, b. E')`
    A (B) {#C}
    :   A, b. E 
`;

test('A. B [{#T}](D). C.')`
    A. B [{#T}](D). C.
`;