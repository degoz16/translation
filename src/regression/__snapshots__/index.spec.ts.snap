// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration ## Start {#your-ideas}{% else %} end 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Start <x ctype=\\"liquid_Attributes\\" equiv-text=\\"{#your-ideas}\\" /><x ctype=\\"liquid_Else\\" equiv-text=\\"{% else %}\\" /> end</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration ## Start {#your-ideas}{% else %} end 2`] = `"## Start {#your-ideas}{% else %} end"`;

exports[`integration (a **18. B**). 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>(a <x ctype=\\"strong_open\\" equiv-text=\\"**\\" />18.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>B<x ctype=\\"strong_close\\" equiv-text=\\"**\\" />).</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration (a **18. B**). 2`] = `"(a **18. B**)."`;

exports[`integration \`{{ var }}\` — one. Two 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source><x ctype=\\"liquid_Variable\\" equiv-text=\\"{{ var }}\\" /><x ctype=\\"code_close\\" equiv-text=\\"\`\\" /> — one.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Two</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration \`{{ var }}\` — one. Two 2`] = `"\`{{ var }}\` — one. Two"`;

exports[`integration A ![](link) b. C. 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>A <g ctype=\\"image\\" equivText=\\"![{{text}}](link)\\" x-begin=\\"![\\" x-end=\\"](link)\\"></g> b.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>C.</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration A ![](link) b. C. 2`] = `"A ![](link) b. C."`;

exports[`integration A (B) {#C}
:   A, b. E 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>A (B) <x ctype=\\"liquid_Attributes\\" equiv-text=\\"{#C}\\" />
:   A, b.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>E</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration A (B) {#C}
:   A, b. E 2`] = `
"A (B) {#C}
:   A, b. E"
`;

exports[`integration A. B [{#T}](D). C. 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>A. B <g ctype=\\"link\\" equivText=\\"[{#T}{{text}}](D)\\" x-begin=\\"[{#T}\\" x-end=\\"](D)\\"></g>.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>C.</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration A. B [{#T}](D). C. 2`] = `"A. B [{#T}](D). C."`;

exports[`integration computes valid sentenses 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Записывает или удаляет элементы из таблиц.</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Один вызов может записать до 16 Мб данных, что может включать до 25 запросов на размещение или удаление.</source>
      </trans-unit>
      <trans-unit id=\\"3\\">
        <source>Каждый элемент может иметь размер до 400 Кб.</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration computes valid sentenses 2`] = `
"Записывает или удаляет элементы из таблиц.
Один вызов может записать до 16 Мб данных, что может включать до 25 запросов на размещение или удаление. Каждый элемент может иметь размер до 400 Кб."
`;

exports[`integration handles autolinks 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>A <x ctype=\\"link_autolink\\" equiv-text=\\"&lt;http://gitgrimbo.github.io/harviewer/master/&gt;\\" /></source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>B <x ctype=\\"link_autolink\\" equiv-text=\\"&lt;https://toolbox.googleapps.com/apps/har_analyzer/?lang=ru&gt;\\" /></source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles autolinks 2`] = `
"* A <http://gitgrimbo.github.io/harviewer/master/>
* B <https://toolbox.googleapps.com/apps/har_analyzer/?lang=ru>"
`;

exports[`integration handles blockquotes 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>1</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>2</source>
      </trans-unit>
      <trans-unit id=\\"3\\">
        <source>3</source>
      </trans-unit>
      <trans-unit id=\\"4\\">
        <source>4</source>
      </trans-unit>
      <trans-unit id=\\"5\\">
        <source>5</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles blockquotes 2`] = `
"> 1
> 2
>
> > 3
> > 4
>
> 5"
`;

exports[`integration handles code regexps with escape chars 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Поле</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Описание</source>
      </trans-unit>
      <trans-unit id=\\"3\\">
        <source>zone</source>
      </trans-unit>
      <trans-unit id=\\"4\\">
        <source>[.]\\\\|[a-z0-9][-a-z0-9.]*\\\\.</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles code regexps with escape chars 2`] = `
"Поле | Описание
--- | ---
zone |  \`\`[.]\\\\|[a-z0-9][-a-z0-9.]*\\\\.\`\`"
`;

exports[`integration handles inline code 1 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Параметр</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Описание</source>
      </trans-unit>
      <trans-unit id=\\"3\\">
        <source>InternalServerError</source>
      </trans-unit>
      <trans-unit id=\\"4\\">
        <source>Произошла</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles inline code 1 2`] = `
"Параметр | Описание
----- | -----
\`InternalServerError\` | Произошла"
`;

exports[`integration handles inline code 3 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Описание значений полей:</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Название поля</source>
      </trans-unit>
      <trans-unit id=\\"3\\">
        <source>Тип</source>
      </trans-unit>
      <trans-unit id=\\"4\\">
        <source>Описание</source>
      </trans-unit>
      <trans-unit id=\\"5\\">
        <source>column_name</source>
      </trans-unit>
      <trans-unit id=\\"6\\">
        <source>string<x ctype=\\"code_close\\" equiv-text=\\"\`\\" /><x ctype=\\"html_inline\\" equiv-text=\\"&lt;br&gt;\\" /><x ctype=\\"code_open\\" equiv-text=\\"\`\\" />required</source>
      </trans-unit>
      <trans-unit id=\\"7\\">
        <source>Имя колонки для TTL</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles inline code 3 2`] = `
"Описание значений полей:
| **Название поля** | **Тип** | **Описание** |
| --- | --- | --- |
|column_name|\`string\`<br>\`required\`|Имя колонки для TTL|"
`;

exports[`integration handles links in brackets 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>(see <g ctype=\\"link\\" equivText=\\"[{{text}}](#nestedblock--timeouts)\\" x-begin=\\"[\\" x-end=\\"](#nestedblock--timeouts)\\">below for nested schema</g>)</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles links in brackets 2`] = `"(see [below for nested schema](#nestedblock--timeouts))"`;

exports[`integration handles links with brackets 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>\\\\[Упр\\\\]</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles links with brackets 2`] = `"[^\\\\[Упр\\\\]^](#example)"`;

exports[`integration handles multiline inline code 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>ALL_CONCAT( expression [ , separator ]
    [ FIXED ... | INCLUDE ... | EXCLUDE ... ]
    [ BEFORE FILTER BY ... ]
)</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles multiline inline code 2`] = `
"\`ALL_CONCAT( expression [ , separator ]
    [ FIXED ... | INCLUDE ... | EXCLUDE ... ]
    [ BEFORE FILTER BY ... ]
)\`"
`;

exports[`integration handles titleless notes 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Для работы в режиме совместимости с Amazon DynamoDB используйте бессерверную (Serverless) конфигурацию БД.</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration handles titleless notes 2`] = `
"{% note info %}

Для работы в режиме совместимости с Amazon DynamoDB используйте бессерверную (Serverless) конфигурацию БД.

{% endnote %}"
`;

exports[`integration link variable leak 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>Request</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>method</source>
      </trans-unit>
      <trans-unit id=\\"3\\">
        <source>url</source>
      </trans-unit>
      <trans-unit id=\\"4\\">
        <source>Path parameters</source>
      </trans-unit>
      <trans-unit id=\\"5\\">
        <source>Name</source>
      </trans-unit>
      <trans-unit id=\\"6\\">
        <source>Type</source>
      </trans-unit>
      <trans-unit id=\\"7\\">
        <source>Description</source>
      </trans-unit>
      <trans-unit id=\\"8\\">
        <source>businessId</source>
      </trans-unit>
      <trans-unit id=\\"9\\">
        <source>integer&amp;lt;int64&amp;gt;</source>
      </trans-unit>
      <trans-unit id=\\"10\\">
        <source>Description</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration link variable leak 2`] = `
"## Request

#||| **method** | **url** ||

||
\`\`\`
POST
\`\`\`
| 
\`\`\`
https://api.example.ru/businesses/{businessId}/offer-mappings/update
\`\`\`
|||#

### Path parameters

#||| **Name** | **Type** | **Description** ||

|| businessId | integer&lt;int64&gt; | Description |||#"
`;

exports[`integration recursive merge tokens 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<xliff xmlns=\\"urn:oasis:names:tc:xliff:document:1.2\\" version=\\"1.2\\">
  <file original=\\"markdown.md\\" source-language=\\"ru-RU\\" target-language=\\"en-US\\" datatype=\\"markdown\\">
    <header>
      <skeleton>
        <external-file href=\\"markdown.skl.md\\"></external-file>
      </skeleton>
    </header>
    <body>
      <trans-unit id=\\"1\\">
        <source>title</source>
      </trans-unit>
      <trans-unit id=\\"2\\">
        <source>Sentense <g ctype=\\"link\\" equivText=\\"[{{text}}](file.md &quot;%%%1%%%&quot;)\\" x-begin=\\"[\\" x-end=\\"](file.md &quot;%%%1%%%&quot;)\\">link</g>.</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`integration recursive merge tokens 2`] = `"Sentense [link](file.md \\"title\\")."`;
